tests/test_plugin.py

```diff
@@ -574,6 +574,82 @@ def test_openchannel_hook(node_factory, bitcoind):
         l1.rpc.fundchannel(l2.info['id'], 100001)
 
 
+def test_openchannel_hook_error_handling(node_factory, bitcoind):
+    """ l2 uses a plugin that should fatal() crash the node.
+
+    This is because the plugin rejects a channel while
+    also setting a close_to address which isn't allowed.
+    """
+    opts = {'plugin': os.path.join(os.getcwd(), 'tests/plugins/openchannel_hook_accepter.py')}
+    # openchannel_reject_but_set_close_to.py')}
+    l1 = node_factory.get_node()
+    l2 = node_factory.get_node(options=opts,
+                               expect_fail=True,
+                               may_fail=True,
+                               allow_broken_log=True)
+    l1.connect(l2)
+
+    # Get some funds.
+    addr = l1.rpc.newaddr()['bech32']
+    txid = bitcoind.rpc.sendtoaddress(addr, 10)
+    numfunds = len(l1.rpc.listfunds()['outputs'])
+    bitcoind.generate_block(1, txid)
+    wait_for(lambda: len(l1.rpc.listfunds()['outputs']) > numfunds)
+
+    # next fundchannel should fail fatal() for l2
+    with pytest.raises(RpcError, match=r'Owning subdaemon openingd died'):
+        l1.rpc.fundchannel(l2.info['id'], 100004)
+    assert l2.daemon.is_in_log("Plugin rejected openchannel but also set a close_to")
```

Comment:
We have a helper that does all this:

```suggestion
    l1.connect(l2)
    
    # The plugin used by l2 causes an error by setting a close_to and rejecting
    with pytest.raises(RpcError, match=r'Owning subdaemon openingd died'):
        l1.fundchannel(l2, 100004)
    assert l2.daemon.is_in_log("Plugin rejected openchannel but also set a close_to")
```

Reply:
Thx for pointing out, fixed that for all three openchannel hook tests.